inbox/quick-response.xml

```diff
@@ -0,0 +1,212 @@
+
+<example><![CDATA[<message to="rootbot@example.com">
+  <body xml:lang="en">no</body>
+</message>]]></example>
+
+    <p>The sender of the original message, in this example <tt>rootbot@example.com</tt>, checks incoming messages for a <tt>&lt;body&gt;</tt> that only contains the <tt>@value</tt> of the corresponding <tt>&lt;translation&gt;</tt> to see if a response was selected. In this example, the <tt>&lt;body&gt;</tt> matches the <tt>@value</tt> of the english translation for the response "No".</p>
+  </section2>
+
+  <section2 anchor="sending-available-actions" topic="Sending a Set of Available Actions">
+    <p>A message with available actions is sent by including one or more &lt;available-action&gt; elements with distinct <tt>@id</tt>s.</p>
+
+<example><![CDATA[<message to="gitbot@example.com">
+  <body xml:lang="en">New merge request opened by ExampleUser: https://git.example.com/example/mrs/3/</body>
+  <available-action xmlns="]]>&ns;<![CDATA[" id="merge">
+    <label xml:lang="en">Merge Now</label>
+  </available-action>
+</message>]]></example>
+
+    <p>A single message MUST NOT contain multiple <tt>&lt;available-action&gt;</tt> elements with the same values for the <tt>@id</tt> attribute. Labels MUST be unique per id and language.</p>
+    <p>Clients that receive a message containing available actions SHOULD offer UI to select one of the actions.</p>
+  </section2>
+
+  <section2 anchor="selecting-an-action" topic="Selecting an Action">
+    <p>When the user selects an action, their client sends a message containing an <tt>&lt;action-selected&gt;</tt> element which identifies the selected action. The message does not contain a <tt>&lt;body&gt;</tt>.</p>
+
+<example><![CDATA[<message to="gitbot@example.com">
+  <action-selected xmlns="]]>&ns;<![CDATA[" id="merge" />
+</message>]]></example>
+  </section2>
+</section1>
+
+<section1 anchor="rules" topic="Business Rules">
+  <p>All message bodies SHOULD always list the (internationalized) allowed responses too, so that users of clients that don't support &xepname; can still know what the allowed responses are.</p>
+  <p>Bots or other entities that indicate allowed responses are free to accept variations of the response values for convenience, for example ignoring the casing or accepting abbreviations of response values.</p>
+  <p>Actions SHOULD only be a quicker way to access a feature that could also be accessed using information in the message body. For example, a bot that notifies about a new merge request includes in its notification message body a link to the web interface where manual merging is possible. An action could offer a more convenient way to merge, without taking the route via the web interface. In summary, users of clients that don't support &xepname; SHOULD still have a way to manually trigger the action.</p>
+  <p>Clients MUST only provide quick responses for the most recently received message that contains text content.</p>
```

Comment:
Why is this? I feel like I should be able to scroll back and respond to older messages too personally, but I haven't really thought it through.

Reply:
This comment is rather non-editorial, maybe we can keep this out of the repo :)

Reply:
@SamWhited I'll ping you in xsf@ to discuss this when I'm done with the editorial fixes

Reply:
I'm not in xsf@; too much traffic and notifications. Let's just discuss it here, if others want to have a more indepth discussion somewhere else they're welcome to.

Reply:
Okay so the reason for this rule is that we don't have clean references. Say, a bot sends a message with a list of allowed responses, 10 random off-topic messages are sent and then you type `yes`. It's obviously confusing and non-obvious that the `yes` belongs to the question the bot asked 10 messages above. Also the bot has no way to know that the other 10 off-topic messages were not attempts to answer its question. All in all, when discussing this we agreed that limiting responses to only the most recent message leads to the least confusion possible.

Reply:
Could it not reference the message ID or something? This restriction seems like it would make the XEP almost useless for any of the use cases I was thinking (eg. things like polls).

Reply:
The point of the allowed responses is that they are exactly equivalent to a "normal" text response, no magic. Manually typing "yes" should be 100% equivalent to clicking the "yes" quick response button. You would most probably use an action to implement a poll.

Reply:
Fair enough. Mandating client behavior here still feels odd to me, because I have a feeling this will be used for something other than bots at some point where we don't care which message the user is responding too, but I see your point and don't have a better suggestion for how to make the fallback work.